IGC/Compiler/Optimizer/OpenCLPasses/AlignmentAnalysis/AlignmentAnalysis.cpp

```diff
@@ -440,9 +444,13 @@ void AlignmentAnalysis::SetInstAlignment(MemSetInst& I)
 #if LLVM_VERSION_MAJOR == 4
     unsigned alignment = iSTD::Max(I.getAlignment(), getAlignValue(I.getRawDest()));
     I.setAlignment(ConstantInt::get(Type::getInt32Ty(I.getContext()), alignment));
-#elif LLVM_VERSION_MAJOR >= 7
+#elif LLVM_VERSION_MAJOR >= 7 && LLVM_VERSION_MAJOR < 14
     unsigned alignment = iSTD::Max(I.getDestAlignment(), getAlignValue(I.getRawDest()));
     I.setDestAlignment(alignment);
+#elif LLVM_VERSION_MAJOR >= 14
+    uint64_t alignment_value = iSTD::Max(I.getDestAlign()->value(), getAlignValue(I.getRawDest()));
+    llvm::Align alignment = llvm::Align(alignment_value);
+    I.setDestAlignment(alignment);
 #endif
```

Comment:
You can drop LLVM 4 support here and leave:

```cpp
#if LLVM_VERSION_MAJOR >= 14
    uint64_t alignment_value = iSTD::Max(I.getDestAlign()->value(), getAlignValue(I.getRawDest()));
    llvm::Align alignment = llvm::Align(alignment_value);
    I.setDestAlignment(alignment);
#else
    unsigned alignment = iSTD::Max(I.getDestAlignment(), getAlignValue(I.getRawDest()));
    I.setDestAlignment(alignment);
#endif
```

From brief analysis: this is a subject to refactor to remove `#if` guard using this [`getAlign()`](https://github.com/intel/intel-graphics-compiler/blob/master/IGC/WrapperLLVM/include/llvmWrapper/Support/Alignment.h#L73) wrapper and adding whatever is necessary for LLVM 14, from the looks of it wrapper on `I.getDestAlign()`/`I.getAlignment()`.

`llvmWrapper/Support/Alignment.h` provides tools to deal with alignment problems between LLVM versions.

Reply:
> From brief analysis: this is a subject to refactor to remove `#if` guard using this [`getAlign()`](https://github.com/intel/intel-graphics-compiler/blob/master/IGC/WrapperLLVM/include/llvmWrapper/Support/Alignment.h#L73) wrapper and adding whatever is necessary for LLVM 14, from the looks of it wrapper on `I.getDestAlign()`/`I.getAlignment()`.

I think I will leave the refactor to some other people since I am not very familiar with this codebase.